### PR TITLE
Document removal of DH_anon (based on drafts from garrett and beurdou…

### LINKED ISSUE #233: Use "raw public keys" instead of DH_anon?

### PR DESCRIPTION
**Pull request by ekr**
…che). Fixes #233

### CODE REVIEW

File: draft-ietf-tls-tls13.md [hunk 1]
```diff
@@ -3753,32 +3756,35 @@ Note: In the case of the CCM mode of AES, two variations exist: "CCM_8" which
...
+
+- Raw public keys {{RFC7250}}.
+- Using a public key contained in a certificate but without
+  validation of the certificate chain or any of its contents.
+
+Either technique used alone is are vulnerable to man-in-the-middle attacks
+and therefore unsafe for general use. However, it is also possible to
+bind such connections to an external authentication mechanism via
+out-of-band validation of the server's public key, trust on first
+use, or channel bindings {{RFC5929}} [[NOTE: TLS 1.3 needs a new
```
Comment (davegarrett) [inline draft-ietf-tls-tls13.md]: missing period after ref for end of sentence

File: draft-ietf-tls-tls13.md [hunk 2]
```diff
@@ -3847,11 +3853,10 @@ Users should be able to view information about the certificate and root CA.
 
 TLS supports a range of key sizes and security levels, including some that
 provide no or minimal security. A proper implementation will probably not
-support many cipher suites. For instance, anonymous Diffie-Hellman is strongly
-discouraged because it cannot prevent man-in-the-middle attacks. Applications
-should also enforce minimum and maximum key sizes. For example, certificate
-chains containing keys or signatures weaker than 2048-bit RSA or 224-bit ECDSA
-are not appropriate for secure applications.
+support many cipher suites. Applications should also enforce minimum and
```
Comment (davegarrett) [inline draft-ietf-tls-tls13.md]: Probably should uppercase the "should" here..

File: draft-ietf-tls-tls13.md [hunk 3]
```diff
@@ -117,6 +119,7 @@ informative:
   I-D.ietf-tls-negotiated-ff-dhe:
 
 
+
```
Comment (davegarrett) [inline draft-ietf-tls-tls13.md]: nit: Intentional new newline? The normative list only has one; don't need three.

File: draft-ietf-tls-tls13.md [hunk 4]
```diff
@@ -3753,32 +3756,35 @@ Note: In the case of the CCM mode of AES, two variations exist: "CCM_8" which
...
+  validation of the certificate chain or any of its contents.
+
+Either technique used alone is are vulnerable to man-in-the-middle attacks
+and therefore unsafe for general use. However, it is also possible to
+bind such connections to an external authentication mechanism via
+out-of-band validation of the server's public key, trust on first
+use, or channel bindings {{RFC5929}} [[NOTE: TLS 1.3 needs a new
+channel binding definition that has not yet been defined.]]
+If no such mechanism is used, then the connection has no protection
+against active mon-in-the-middle attack; applications MUST NOT use TLS
```
Comment (davegarrett) [inline draft-ietf-tls-tls13.md]: 1) These are really just two sentences. You can just drop the semicolon and replace it with a period and subsequent capitalization.
2) "mon-in-the-middle" -> "man-in-the-middle"

**Comment by davegarrett** [comment]
Aside from the little nits and typo, the text itself looks good to me.

File: draft-ietf-tls-tls13.md (hunk 2 above)
Comment (ekr) [inline draft-ietf-tls-tls13.md]: Sure, though this is old text.

File: draft-ietf-tls-tls13.md (hunk 4 above)
Comment (ekr) [inline draft-ietf-tls-tls13.md]: These are closely related, so I prefer the ';'

**Pull request merged**